Reachable unreachable panic in `sys_timerfd_settime` when invalid flag given

### Describe the bug
There is a reachable unreachable panic in `sys_timerfd_settime()` at kernel/src/syscall/timerfd_settime.rs:57 when call `timerfd_settime` with invalid flag.


https://github.com/asterinas/asterinas/blob/e9f35f6ab8c5be1298ad197941c0305934770588/kernel/src/syscall/timerfd_settime.rs#L57

### To Reproduce
1. Compile the program and run:
```C
#define _GNU_SOURCE
#include <stdio.h>
#include <stdlib.h>
#include <sys/timerfd.h>
#include <string.h>
#include <unistd.h>
#include <stdint.h>
#include <time.h>
#include <errno.h>

int main() {
    int fd = timerfd_create(CLOCK_MONOTONIC, 0);
    if (fd == -1) {
        perror("timerfd_create failed");
        return 1;
    }

    struct itimerspec new_value;
    memset(&new_value, 0, sizeof(new_value));
    new_value.it_value.tv_sec = 1;
    new_value.it_value.tv_nsec = 0;
    new_value.it_interval.tv_sec = 0;
    new_value.it_interval.tv_nsec = 0;

    // int invalid_flags = 99;
    int invalid_flags = 3;

    if (timerfd_settime(fd, invalid_flags, &new_value, NULL) == -1) {
        perror("timerfd_settime failed");
    } else {
        printf("timerfd_settime succeeded unexpectedly\n");
    }

    close(fd);
    return 0;
}
```

### Expected behavior
Asterinas shall not panic when run the program.

### Environment
- Official docker asterinas/asterinas:0.15.0-20250516
- Intel(R) Xeon(R) Gold 6230R CPU @ 2.10GHz
- Asterinas version: main e9f35f6

### Log
```
~ # /root/timefd.c 
[   204.803] ERROR: Uncaught panic:
        internal error: entered unreachable code
        at /root/asterinas/kernel/src/syscall/timerfd_settime.rs:57
        on CPU 0 by thread Some(Thread { task: (Weak), data: Any { .. }, status: AtomicThreadStatus(1), cpu_affinity: AtomicCpuSet { bits: [1] }, sched_attr: SchedAttr { policy: SchedPolicyState { kind: AtomicSchedPolicyKind(2), policy: UnsafeCell { .. } }, last_cpu: AtomicCpuId(0), real_time: RealTimeAttr { prio: 99, time_slice: 31489680 }, fair: FairAttr { weight: 1024, vruntime: 406294645 } } })
Printing stack trace:
   1: fn 0xffffffff887daa60 - pc 0xffffffff887daaaa / registers:

     rax                0x8; rdx           0x200202; rcx 0xffffdfffffdcb1d0; rbx                0x0;
     rsi 0xffffdfffffdcb240; rdi 0xffffffff8881fdc1; rbp                0x0; rsp 0xffffdfffffdcb2c0;

   2: fn 0xffffffff884fcd60 - pc 0xffffffff884fd509 / registers:

     rax                0x8; rdx           0x200202; rcx 0xffffdfffffdcb1d0; rbx                0x0;
     rsi 0xffffdfffffdcb240; rdi 0xffffffff8881fdc1; rbp                0x0; rsp 0xffffdfffffdcb310;

   3: fn 0xffffffff884fcd50 - pc 0xffffffff884fcd5a / registers:

     rax                0x8; rdx           0x200202; rcx 0xffffdfffffdcb1d0; rbx                0x0;
     rsi 0xffffdfffffdcb240; rdi 0xffffffff8881fdc1; rbp                0x0; rsp 0xffffdfffffdcb720;

   4: fn 0xffffffff8804b000 - pc 0xffffffff8804b00a / registers:

     rax                0x8; rdx           0x200202; rcx 0xffffdfffffdcb1d0; rbx                0x0;
     rsi 0xffffdfffffdcb240; rdi 0xffffffff8881fdc1; rbp                0x0; rsp 0xffffdfffffdcb730;

   5: fn 0xffffffff8893a790 - pc 0xffffffff8893a7d4 / registers:

     rax                0x8; rdx           0x200202; rcx 0xffffdfffffdcb1d0; rbx                0x0;
     rsi 0xffffdfffffdcb240; rdi 0xffffffff8881fdc1; rbp                0x0; rsp 0xffffdfffffdcb740;

   6: fn 0xffffffff8893a840 - pc 0xffffffff8893a880 / registers:

     rax                0x8; rdx           0x200202; rcx 0xffffdfffffdcb1d0; rbx                0x0;
     rsi 0xffffdfffffdcb240; rdi 0xffffffff8881fdc1; rbp                0x0; rsp 0xffffdfffffdcb780;

   7: fn 0xffffffff881060e0 - pc 0xffffffff881069aa / registers:

     rax                0x8; rdx           0x200202; rcx 0xffffdfffffdcb1d0; rbx                0x0;
     rsi 0xffffdfffffdcb240; rdi 0xffffffff8881fdc1; rbp                0x0; rsp 0xffffdfffffdcb7f0;

   8: fn 0xffffffff88476930 - pc 0xffffffff884a22cd / registers:

     rax                0x8; rdx           0x200202; rcx 0xffffdfffffdcb1d0; rbx                0x0;
     rsi 0xffffdfffffdcb240; rdi 0xffffffff8881fdc1; rbp                0x0; rsp 0xffffdfffffdcbc20;

   9: fn 0xffffffff88408510 - pc 0xffffffff884085a1 / registers:

     rax                0x8; rdx           0x200202; rcx 0xffffdfffffdcb1d0; rbx                0x0;
     rsi 0xffffdfffffdcb240; rdi 0xffffffff8881fdc1; rbp                0x0; rsp 0xffffdfffffddd510;

  10: fn 0xffffffff885c50e0 - pc 0xffffffff885c58ba / registers:

     rax                0x8; rdx           0x200202; rcx 0xffffdfffffdcb1d0; rbx                0x0;
     rsi 0xffffdfffffdcb240; rdi 0xffffffff8881fdc1; rbp                0x0; rsp 0xffffdfffffddd690;

  11: fn 0xffffffff880750d0 - pc 0xffffffff880750d6 / registers:

     rax                0x8; rdx           0x200202; rcx 0xffffdfffffdcb1d0; rbx                0x0;
     rsi 0xffffdfffffdcb240; rdi 0xffffffff8881fdc1; rbp                0x0; rsp 0xffffdfffffdddd00;

  12: fn 0xffffffff8819e180 - pc 0xffffffff8819e197 / registers:

     rax                0x8; rdx           0x200202; rcx 0xffffdfffffdcb1d0; rbx                0x0;
     rsi 0xffffdfffffdcb240; rdi 0xffffffff8881fdc1; rbp                0x0; rsp 0xffffdfffffdddd10;

  13: fn 0xffffffff8819e110 - pc 0xffffffff8819e131 / registers:

     rax                0x8; rdx           0x200202; rcx 0xffffdfffffdcb1d0; rbx                0x0;
     rsi 0xffffdfffffdcb240; rdi 0xffffffff8881fdc1; rbp                0x0; rsp 0xffffdfffffdddd30;

  14: fn 0xffffffff88227990 - pc 0xffffffff882279a2 / registers:

     rax                0x8; rdx           0x200202; rcx 0xffffdfffffdcb1d0; rbx                0x0;
     rsi 0xffffdfffffdcb240; rdi 0xffffffff8881fdc1; rbp                0x0; rsp 0xffffdfffffdddd80;

  15: fn 0xffffffff880717e0 - pc 0xffffffff880717f6 / registers:

     rax                0x8; rdx           0x200202; rcx 0xffffdfffffdcb1d0; rbx                0x0;
     rsi 0xffffdfffffdcb240; rdi 0xffffffff8881fdc1; rbp                0x0; rsp 0xffffdfffffddddb0;

  16: fn 0xffffffff880aeca0 - pc 0xffffffff880aecac / registers:

     rax                0x8; rdx           0x200202; rcx 0xffffdfffffdcb1d0; rbx                0x0;
     rsi 0xffffdfffffdcb240; rdi 0xffffffff8881fdc1; rbp                0x0; rsp 0xffffdfffffdddf50;

  17: fn 0xffffffff88073b60 - pc 0xffffffff88073b6e / registers:

     rax                0x8; rdx           0x200202; rcx 0xffffdfffffdcb1d0; rbx                0x0;
     rsi 0xffffdfffffdcb240; rdi 0xffffffff8881fdc1; rbp                0x0; rsp 0xffffdfffffdddf80;

  18: fn 0xffffffff887e9160 - pc 0xffffffff887e9185 / registers:

     rax                0x8; rdx           0x200202; rcx 0xffffdfffffdcb1d0; rbx                0x0;
     rsi 0xffffdfffffdcb240; rdi 0xffffffff8881fdc1; rbp                0x0; rsp 0xffffdfffffdddfa0;

  19: fn 0xffffffff887413a0 - pc 0xffffffff88741439 / registers:

     rax                0x8; rdx           0x200202; rcx 0xffffdfffffdcb1d0; rbx                0x0;
     rsi 0xffffdfffffdcb240; rdi 0xffffffff8881fdc1; rbp                0x0; rsp 0xffffdfffffdddfd0;


     rax                0x8; rdx           0x200202; rcx 0xffffdfffffdcb1d0; rbx                0x0;
     rsi 0xffffdfffffdcb240; rdi 0xffffffff8881fdc1; rbp                0x0; rsp 0xffffdfffffdde000;

OSDK] The kernel seems panicked. Parsing stack trace for source lines:
(  1) /root/asterinas/ostd/src/panic.rs:107
(  2) /root/asterinas/kernel/src/thread/oops.rs:129
(  3) ??:?
(  4) 75yx17rn96ufy2vu3zk9y8ez8:?
(  5) ??:?
(  6) /root/.rustup/toolchains/nightly-2025-02-01-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/panicking.rs:217
(  7) /root/asterinas/kernel/src/syscall/timerfd_settime.rs:55
(  8) /root/asterinas/kernel/src/syscall/mod.rs:201
(  9) /root/asterinas/kernel/src/syscall/mod.rs:351
( 10) /root/asterinas/kernel/src/thread/task.rs:83
( 11) /root/.rustup/toolchains/nightly-2025-02-01-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/ops/function.rs:250
( 12) /root/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/unwinding-0.2.5/src/panicking.rs:56
( 13) /root/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/unwinding-0.2.5/src/panicking.rs:42
( 14) /root/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/unwinding-0.2.5/src/panic.rs:87
( 15) /root/asterinas/kernel/src/thread/oops.rs:54
( 16) /root/asterinas/kernel/src/thread/task.rs:109
( 17) /root/.rustup/toolchains/nightly-2025-02-01-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/ops/function.rs:250
( 18) /root/.rustup/toolchains/nightly-2025-02-01-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/alloc/src/boxed.rs:2016
( 19) /root/asterinas/ostd/src/task/mod.rs:234
make: *** [Makefile:230: run] Error 1
```

--- a/kernel/src/syscall/timer_settime.rs
+++ b/kernel/src/syscall/timer_settime.rs
@@ -43,12 +43,10 @@ pub fn sys_timer_settime(
         // Clear previous timer
         timer.cancel();
     } else {
-        let timeout = if flags == 0 {
+        let timeout = if (flags & TIMER_ABSTIME) == 0 {
             Timeout::After(expire_time)
-        } else if flags == TIMER_ABSTIME {
-            Timeout::When(expire_time)
         } else {
-            unreachable!()
+            Timeout::When(expire_time)
         };
         timer.set_timeout(timeout);
     }

--- a/kernel/src/syscall/timerfd_settime.rs
+++ b/kernel/src/syscall/timerfd_settime.rs
@@ -6,7 +6,12 @@ use super::SyscallReturn;
 use crate::{
     fs::file_table::FileDesc,
     prelude::*,
-    time::{itimerspec_t, timer::Timeout, timerfd::TimerfdFile, timespec_t, TIMER_ABSTIME},
+    time::{
+        itimerspec_t,
+        timer::Timeout,
+        timerfd::{TFDSetTimeFlags, TimerfdFile},
+        timespec_t,
+    },
 };
 
 pub fn sys_timerfd_settime(
@@ -16,6 +21,8 @@ pub fn sys_timerfd_settime(
     old_itimerspec_addr: Vaddr,
     ctx: &Context,
 ) -> Result<SyscallReturn> {
+    let flags = TFDSetTimeFlags::from_bits(flags as u32)
+        .ok_or_else(|| Error::with_message(Errno::EINVAL, "invalid flags for timerfd_settime"))?;
     let file_table = ctx.thread_local.borrow_file_table();
     let file_table_locked = file_table.unwrap().read();
     let timerfd_file = file_table_locked.get_file(fd as _)?;
@@ -45,16 +52,15 @@ pub fn sys_timerfd_settime(
     // when the timer is rearmed.
     timerfd_file.clear_ticks();
 
-    const TFD_TIMER_CANCEL_ON_SET: i32 = 2;
     if expire_time != Duration::ZERO {
-        let timeout = if flags == 0 {
-            Timeout::After(expire_time)
-        } else if flags == TIMER_ABSTIME {
-            Timeout::When(expire_time)
-        } else if flags == TFD_TIMER_CANCEL_ON_SET {
+        if flags.contains(TFDSetTimeFlags::TFD_TIMER_CANCEL_ON_SET) {
             unimplemented!()
+        }
+
+        let timeout = if flags.contains(TFDSetTimeFlags::TFD_TIMER_ABSTIME) {
+            Timeout::When(expire_time)
         } else {
-            unreachable!()
+            Timeout::After(expire_time)
         };
         timer.set_timeout(timeout);
     }

--- a/kernel/src/time/timerfd.rs
+++ b/kernel/src/time/timerfd.rs
@@ -27,10 +27,18 @@ pub struct TimerfdFile {
 }
 
 bitflags! {
-    /// The flags useds for timerfd-related operations.
+    /// The flags used for timerfd-related operations.
     pub struct TFDFlags: u32 {
         const TFD_CLOEXEC = CreationFlags::O_CLOEXEC.bits();
         const TFD_NONBLOCK = StatusFlags::O_NONBLOCK.bits();
+    }
+}
+
+bitflags! {
+    /// The flags used for timerfd settime operations.
+    pub struct TFDSetTimeFlags: u32 {
+        const TFD_TIMER_ABSTIME = 0x1;
+        const TFD_TIMER_CANCEL_ON_SET = 0x2;
     }
 }
 

